File: shared/core/src/main/java/com/lacker/micros/utils/encrypt/Md5Utils.java
package com.lacker.micros.utils.encrypt;

import org.springframework.util.DigestUtils;

public enum Md5Utils {

    X;

    public String computeHash(String text) {
        return DigestUtils.md5DigestAsHex(text.getBytes()).toUpperCase();
    }

    public boolean verify(String text, String hash) {
        return computeHash(text).equalsIgnoreCase(hash);
    }
}

File: library/core/src/main/kotlin/com/mlacker/micros/utils/encrypt/Md5Utils.kt
package com.mlacker.micros.utils.encrypt

import org.springframework.util.DigestUtils

enum class Md5Utils {

    X;

    fun computeHash(text: String) = DigestUtils.md5DigestAsHex(text.toByteArray()).toUpperCase()

    fun verify(text: String, hash: String) = computeHash(text).equals(hash, ignoreCase = true)
}